app/frontend/state.ts

```diff
@@ -12,112 +12,123 @@ export interface SendTransactionSummary {
 }
 
 export interface State {
+  // general
   loading: boolean
   loadingMessage: string
   alert: any // TODO
-  displayWelcome: boolean
-  displayInfoModal: boolean
   currentTab: 'wallet-info'
-  walletIsLoaded: boolean
-  shouldShowStakingBanner: boolean
+  sendSentry: {
+    event?: any
+    resolve?: (shouldSend: boolean) => void
+  }
   errorBannerContent: string
-  sendAddress: any // TODO
-  sendAmount: any // TODO
-  keepConfirmationDialogOpen: boolean
+  shouldShowUnexpectedErrorModal: boolean
+  error?: any
+  activeMainTab: MainTabs
+  shouldShowContactFormModal?: boolean
+  shouldShowExportOption?: boolean
+  conversionRates?: {data: {USD: number; EUR: number}}
 
-  sendTransactionSummary: SendTransactionSummary
+  // cache
+  displayWelcome: boolean
+  shouldShowStakingBanner: boolean
+  displayInfoModal: boolean
+  shouldShowPremiumBanner?: boolean
 
-  router: {
-    pathname: string
-    hash: string
-  }
+  // login / logout
+  autoLogin: boolean
+  authMethod: AuthMethod
+  shouldShowDemoWalletWarningDialog: boolean
+  logoutNotificationOpen: boolean
+  walletIsLoaded: boolean
+  isShelleyCompatible: any
+  shouldShowNonShelleyCompatibleDialog: any
```

Comment:
If it doesn't require additional extensive changes, I'd type these to boolean although it's out slightly of scope, but you've delved into it already so I think it's safe to do

Same goes for another obvious booleans and simple types, for example calculatingDelegationFee, isDelegationValid, etc.

Reply:
Yeah I know, but as you mentioned, we can type these when we're gonna type the components that work with them. By this logic I could type the whole state :smile: